runtime/src/iree/task/executor.c

```diff
@@ -492,6 +472,11 @@ static iree_task_t* iree_task_executor_try_steal_task_from_affinity_set(
     worker_index += offset + 1;
     mask = iree_shr(mask, offset + 1);
     iree_task_worker_t* victim_worker = &executor->workers[victim_index];
+    if (iree_atomic_load_int32(&victim_worker->state,
```

Comment:
The unfortunate thing here is that we now need to touch each worker from all other workers - the victim mask was allowing us to perform up to 64 comparisons with a single operation. This is fairly hot code (~millions, runs after every bit of work on each worker) so it doesn't feel great to move this near-guaranteed cache miss and conflict into the hot loop of the hot path, but honestly it may all be in the noise anyway compared to matmuls.

If you can think of anything that avoids this, though, that'd be great. One of the goals of the original bit masks was to avoid needing to traverse all of the workers like this - if we have to we have to, but if we don't we definitely shouldn't. I still think this cleanup is worth it just trying to see if we can work out a compromise :)